### PR TITLE
Add retry options for starting workflow via CLI (#3289)

### PR DESCRIPTION
**Pull request by vytautas-karpavicius**
This change exposes new optional CLI parameters allowing to set retry policy when starting new workflows.

**What changed?**
Added new optional parameters when starting workflow:
- `--retry_expiration`
- `--retry_attempts`
- `--retry_interval`
- `--retry_backoff`
- `--retry_max_interval`

**Why?**
Retry policy was not exposed via CLI before when starting a workflow.

**How did you test it?**
Tested with locally built CLI and local server these scenarios:

**No retry options** 
```
./cadence --domain mickey workflow start --tasklist mini --workflow_type pluto --execution_timeout 1
```
No retry policy was set.

 **Retries limited by max attempts** 
```
./cadence --domain mickey workflow start --tasklist mini --workflow_type pluto --execution_timeout 1 --retry_attempts 5
```
Retry policy set to:
```
retryPolicy.initialIntervalInSeconds 10
retryPolicy.backoffCoefficient       1
retryPolicy.maximumAttempts          5
```

 **Retries limited by expiration** 
```
./cadence --domain mickey workflow start --tasklist mini --workflow_type pluto --execution_timeout 1 --retry_expiration 20
```
Retry policy set to:
```
retryPolicy.initialIntervalInSeconds    10
retryPolicy.backoffCoefficient          1
retryPolicy.expirationIntervalInSeconds 20
```

**Retries with exponential backoff**
```
./cadence --domain mickey workflow start --tasklist mini --workflow_type pluto --execution_timeout 1 --retry_expiration 200 --retry_backoff=2 --retry_interval=1 --retry_max_interval=16
```
Retry policy set to:
```
retryPolicy.initialIntervalInSeconds    1
retryPolicy.backoffCoefficient          2
retryPolicy.maximumIntervalInSeconds    16
retryPolicy.expirationIntervalInSeconds 200
```

**Potential risks**

### CODE REVIEW

File: tools/cli/workflowCommands.go [hunk 1]
```diff
@@ -203,6 +203,15 @@ func startWorkflowHelper(c *cli.Context, shouldPrintProgress bool) {
 	if c.IsSet(FlagCronSchedule) {
 		startRequest.CronSchedule = common.StringPtr(c.String(FlagCronSchedule))
 	}
+	if c.IsSet(FlagRetryAttempts) {
+		interval := int32(c.Int(FlagRetryInterval))
+		attempts := int32(c.Int(FlagRetryAttempts))
+		startRequest.RetryPolicy = &s.RetryPolicy{
+			MaximumAttempts:          common.Int32Ptr(attempts),
```
Comment (yycptt) [inline tools/cli/workflowCommands.go:210]: I think in some use cases user may want to retry a workflow for certain period of time and don't care much about the exact number of attempts. In that case, `ExpirationInterval` might be more useful.

File: tools/cli/workflowCommands.go [hunk 2]
```diff
@@ -203,6 +203,15 @@ func startWorkflowHelper(c *cli.Context, shouldPrintProgress bool) {
...
+			InitialIntervalInSeconds: common.Int32Ptr(interval),
+			BackoffCoefficient:       common.Float64Ptr(1.0), // no backoff
```
Comment (yycptt) [inline tools/cli/workflowCommands.go:212]: Any reason we don't want to allow user to specify this value?

File: tools/cli/flags.go [hunk 3]
```diff
@@ -226,6 +226,8 @@ const (
 	FlagFailoverTypeWithAlias             = FlagFailoverType + ", ft"
 	FlagFailoverTimeout                   = "failover_timeout_seconds"
 	FlagFailoverTimeoutWithAlias          = FlagFailoverTimeout + ", fts"
+	FlagRetryInterval                     = "retry_interval"
```
Comment (vancexu) [inline tools/cli/flags.go:229]: maybe explicitly make it `retry_interval_in_sec` and add a alias?

File: tools/cli/flags.go [hunk 4]
```diff
@@ -354,6 +356,15 @@ func getFlagsForStart() []cli.Flag {
 				"If value is array, use json array like [\"a\",\"b\"], [1,2], [\"true\",\"false\"], [\"2019-06-07T17:16:34-08:00\",\"2019-06-07T18:16:34-08:00\"]. " +
 				"Use 'cluster get-search-attr' cmd to list legal keys and value types",
 		},
+		cli.IntFlag{
+			Name:  FlagRetryAttempts,
+			Usage: "Optional retry attempts. If set workflow will be retried the specified amount of times.",
```
Comment (vancexu) [inline tools/cli/flags.go:361]: maybe add to usage: "...  with fixed retry interval"

File: tools/cli/workflowCommands.go (hunk 1 above)
Comment (vytautas-karpavicius) [reply tools/cli/workflowCommands.go]: Agreed, this could be useful. Updated code.

File: tools/cli/workflowCommands.go (hunk 2 above)
Comment (vytautas-karpavicius) [reply tools/cli/workflowCommands.go]: No specific reason, other than tried to keep CLI simpler.
Updated code to include it as well.

File: tools/cli/flags.go (hunk 3 above)
Comment (vytautas-karpavicius) [reply tools/cli/flags.go:229]: Other options don't seem to follow this convention, e.g. `execution_timeout` or `context_timeout` don't have explicit name. There are some that have `_seconds` suffix like `failover_timeout_seconds`. This makes me hesitant to add this third style `_in_sec`.   It is also somewhat more lengthy to type.

File: tools/cli/flags.go (hunk 4 above)
Comment (vytautas-karpavicius) [reply tools/cli/flags.go:368]: Either this or additional backoff option. For now added backoff option as suggested by @yycptt.

**Pull request merged**